etl/steps/data/meadow/hmd/2023-09-19/hmd.py

```diff
@@ -0,0 +1,267 @@
+    # Initiate empty list of tables
+    tbs = []
+    # Get files in folder
+    files = glob(os.path.join(path, "*.txt"))
+    log.info(f"Creating dataframe from available files in {path}...")
+    # Load and parse each txt file
+    for f in files:
+        # Parse txt -> table
+        tb = make_tb_from_txt_callable(f)
+        tbs.append(tb)
+    tb = pr.concat(tbs, ignore_index=True)
+    return tb
+
+
+def _make_tb_from_txt_lt(txt_path: Path | str) -> Table:
+    """Load and parse a txt file into a Table."""
+    # Regex to parse TXT file
+    FILE_REGEX = (
+        r"(?P<country>[a-zA-Z\-\s,]+), Life tables \((?P<type>[a-zA-Z]+) (?P<format>\d+x\d+)\), (?P<sex>[a-zA-Z]+)"
+        r"\tLast modified: (?P<last_modified>\d+ [a-zA-Z]{3} \d+);  Methods Protocol: v\d+ \(\d+\)\n\n(?P<data>(?s:.)*)"
+    )
+    # Read single file
+    with open(txt_path, "r") as f:
+        text = f.read()
+    # Get relevant fields
+    match = re.search(FILE_REGEX, text)
+    if match is not None:
+        groups = match.groupdict()
+    else:
+        raise ValueError(f"No match found in {f}! Please revise that source files' content matches FILE_REGEX.")
+    # Build df
+    tb_str = groups["data"].strip()
+    tb_str = re.sub(r"\n\s+", "\n", tb_str)
+    tb_str = re.sub(r"[^\S\r\n]+", "\t", string=tb_str)
+    tb = pr.read_csv(StringIO(tb_str), sep="\t")
+    # Add dimensions
+    tb = tb.assign(
+        country=groups["country"],
+        sex=groups["sex"],
+        type=groups["type"],
+        format=groups["format"],
+    )
+    return tb
+
+
+def _make_tb_from_txt_exp(txt_path: Path | str) -> Table:
+    """Load and parse a txt file into a Table."""
+    # Regex to parse TXT file
+    FILE_REGEX = (
+        r"(?P<country>[a-zA-Z\-\s,]+), Exposure to risk \((?P<type>[a-zA-Z]+) (?P<format>\d+x\d+)\),\s\tLast modified: "
+        r"(?P<last_modified>\d+ [a-zA-Z]{3} \d+);  Methods Protocol: v\d+ \(\d+\)\n\n(?P<data>(?s:.)*)"
+    )
+    # Read single file
+    with open(txt_path, "r") as f:
+        text = f.read()
+    # Get relevant fields
+    match = re.search(FILE_REGEX, text)
+    if match is not None:
+        groups = match.groupdict()
+    else:
+        raise ValueError(f"No match found in {f}! Please revise that source files' content matches FILE_REGEX.")
+    # Build df
+    tb_str = groups["data"].strip()
+    tb_str = re.sub(r"\n\s+", "\n", tb_str)
+    tb_str = re.sub(r"[^\S\r\n]+", "\t", string=tb_str)
+    tb = pr.read_csv(StringIO(tb_str), sep="\t")
+    # Melt
+    tb = tb.melt(id_vars=["Age", "Year"], var_name="sex", value_name="exposure")
+    # Add dimensions
+    tb = tb.assign(
+        country=groups["country"],
+        type=groups["type"],
+        format=groups["format"],
+    )
+    return tb
+
+
+def proces_missing_data_lt(tb: Table) -> Table:
+    """Check and process missing data.
+
+    Missing data comes as '.', instead replace these with NaNs.
+    """
+    # Expected missing data
+    frac_rows_missing_expected = 0.01
```

Comment:
Maybe good to set these hard-coded values at the top of the file so it's easier to revisit them when running an update?